src/core/environment.go

```diff
@@ -112,6 +113,33 @@ func (env *EnvironmentType) NewHostInfo(agentVersion string, tags *[]string, con
 	return env.host
 }
 
+/*  Returns string containing
+	   sysname	 Name of the operating system implementation.
+	   nodename	 Network name of this machine.
+	   release	 Release level of the operating	system.
+	   version	 Version level of the operating	system.
+	   machine	 Machine hardware platform.
+
+	Different platforms have different Utsname struct definition.
+	https://cs.opensource.google/search?q=utsname&ss=go%2Fx%2Fsys&start=11
+
+	TODO :- Make this function platform agnotic to pull uname (uname -a).
+*/
+func (env *EnvironmentType) GetUnixName() string {
+	var utsname unix.Utsname
+	err := unix.Uname(&utsname)
+	unixName := ""
+	if(err == nil) {
```

Comment:
The convention in golang is to handle the error path in the if statement.

if err != nil {
  // handle err
}

do the rest of the code workflow

return results.

Reply:
Fixed.